torch/_export/utils.py

```diff
@@ -617,3 +618,59 @@ def _extract_pytree_key(x):
             ):
                 constants[new_name] = constant
                 del constants[name]
+
+
+def runtime_asserts_naming_pass(gm, shape_env, forward_arg_names, strict):
+    import sympy
+
+    if shape_env.dim_constraints is None or forward_arg_names is None:  # no dynamic shapes/names
+        return
+
+    symbol_to_source = shape_env.dim_constraints._dcp.symbol_to_source
+
+    # get actual names for symbols
```

Comment:
@avikchaudhuri do you know if there's a way to tell from the ShapeEnv or DimConstraints if the sources are from Dynamo or not (i.e. strict/non-strict)? For strict we'll get source names like L['x'].size()[0], for non-strict L['args'][0][0].size()[0].

Adding a strict flag is ok for calls to `export()`, but it's nicer if we can run this somewhere we don't have access to that info (e.g. in run_decompositions())